### PR TITLE
User deletion

### PR DESCRIPTION
**Pull request by emileswarts**
This GDPR cleanup used to be done by a lambda.

We've since decided that all cron tasks will be run as ECS scheduled
tasks.  This is the rewrite of it from Python to Ruby.

### CODE REVIEW

File: lib/gdpr/use_case/delete_users.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Gdpr::UseCase::DeleteUsers
```
Comment (kjdchapman) [inline lib/gdpr/use_case/delete_users.rb]: I'd call this `DeleteStaleUsers` or `DeleteOldUsers` - something like that?

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Lint/AmbiguousBlockAssociation:
```
Comment (kjdchapman) [inline .rubocop.yml]: I feel like I ran into this the other day and managed to fix it. I'll have a quick play to see if I can?

File: lib/gdpr/use_case/obfusticate_sponsors.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Gdpr::UseCase::ObfusticateSponsors
```
Comment (kjdchapman) [inline lib/gdpr/use_case/obfusticate_sponsors.rb]: spelling, `Obfusticate` -> `Obfuscate`. This is in many other places too, but I think it's all the same misspelling, so find and replace should hopefully be e-z.

File: lib/gdpr/gateway/usedetails.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    DB.run('DELETE FROM userdetails
+        WHERE (last_login < DATE_SUB(NOW(), INTERVAL 12 MONTH)
+        OR (last_login IS NULL AND created_at < DATE_SUB(NOW(), INTERVAL 12 MONTH)))
+        AND username != "HEALTH"')
+  end
+
+  def obfusticate_sponsors
+    DB.run("UPDATE userdetails ud1
+        LEFT JOIN userdetails as ud2 ON ud1.sponsor = ud2.contact
+        SET ud1.sponsor = REPLACE(ud1.sponsor, SUBSTRING_INDEX(ud1.sponsor, '@', '1'), 'user')
```
Comment (kjdchapman) [inline lib/gdpr/gateway/usedetails.rb]: Just to double-check my understanding of the SQL black magic - this replaces `kyle@arbitrary-domain.com` with `user@arbitrary-domain.com`?

File: .rubocop.yml (hunk 2 above)
Comment (kjdchapman) [reply .rubocop.yml]: Ah, I see the issue you referenced in commit message. Alright, this seems fine to me.

File: lib/gdpr/gateway/usedetails.rb (hunk 4 above)
Comment (emileswarts) [reply lib/gdpr/gateway/usedetails.rb]: Yup

File: lib/gdpr/use_case/delete_users.rb (hunk 1 above)
Comment (emileswarts) [reply lib/gdpr/use_case/delete_users.rb]: Fair enough, called it DeleteInactiveUsers

**Pull request merged**